### ISSUE TITLE
it seemed that length function is wrong

### ISSUE THREAD

**Issue by YuBeomGon** [zaps166/libsimplewebm#2]
in example.cpp,
total means that total byte number in file.
available means the left byte number in file.
is it right??

but in below code,
total and available have same value, 

	int Length(long long *total, long long *available)
	{
		if (!m_file)
			return -1;
		const off_t pos = ftell(m_file);
		fseek(m_file, 0, SEEK_END);
		if (total)
			*total = ftell(m_file);
		if (available)
			*available = ftell(m_file);
		fseek(m_file, pos, SEEK_SET);
		return 0;
	}

**Comment by zaps166** [comment]
Looks like a bug, did you test it?

**Comment by keyboardsage** [comment]
Some [source code](https://github.com/keyboardsage/OpenAVMedia/blob/main/tests/test3.cpp) from my project depends on libsimplewebm and I had a similar thought when looking at your example.

It seems unnecessary for Google's webm library to require both total and available as arguments when they are set to the same value. The library could handle this internally, simplifying the user interface.

Additionally, I wasn't sure how to interpret the arguments. Without documentation, there could be two possible interpretations:
1. The total number of bytes is `total`. The `available` is how many bytes are left to play. -OR-
2. The number of bytes played is `total` and the number of bytes available in the file period is `available`
So for kicks, I set `total` and `available` to `total_pos` and `total_pos - curr_pos`, respectively. Then I flipped their values to see what happens in my code below.

```
int Length(long long* total, long long* available)
	{
		if (!m_file) return -1; // if file descriptor/handle is valid...

		const off_t curr_pos = ftell(m_file);  // ...get current position
		fseek(m_file, 0, SEEK_END);
                const off_t total_pos = ftell(m_file); // ...and total file length
                fseek(m_file, curr_pos, SEEK_SET); // maintain the file's original position
		
                // update the parameters
                if (total)
			//*total = total_pos;
                        *total = total_pos - curr_pos;
		if (available)
			//*available = total_pos - curr_pos; // TODO: this seems logical but it causes the video playback to fail
                        *available = total_pos;

		return 0;
	}
```

The result is that In both cases the program failed.
```
$ ./tests/test3 ../tests/assets/big-buck-bunny_trailer.webm 
Failed to open file: ../tests/assets/big-buck-bunny_trailer.webm
```

I didn't take the time to look into the reason for why it fails, because I was satisfied with just knowing that it fails. Finally, I have looked at other project's source code such as [this one](https://github.com/ssincak/webmPlayer/blob/master/src/file_reader.cpp) and they were always written the same way as in libsimplewebm with both `total` and `available` set to the same value.

TLDR; main takeaway is I don't know the root cause for **why** `total` and `available` need to be set to the same value; but my empirical tests confirmed that they do in order to work, and everyone does it this way also.
